collector/mdadm_linux.go

```diff
@@ -152,6 +152,8 @@ func parseMdstat(mdStatusFilePath string) ([]mdStatus, error) {
 	mdStates := make([]mdStatus, 0, estimateMDs)
 
 	for i, l := range lines {
+		active, total, size = 0, 0, 0
```

Comment:
Agree with @grobie, 0 is really misleading in this case. Using -1 and converting to NaN in caller seems best to me.